libavpipe/src/avpipe_xc.c

```diff
@@ -3392,14 +3396,30 @@ avpipe_xc(
 
     int video_stream_index = decoder_context->video_stream_index;
     if (params->xc_type & xc_video) {
+
+        if (av_cmp_q(decoder_context->format_context->streams[video_stream_index]->r_frame_rate, decoder_context->format_context->streams[video_stream_index]->avg_frame_rate)) {
+            elv_log("frame rate discrepancy r=%d/%d avg=%d/%d",
+                decoder_context->format_context->streams[video_stream_index]->r_frame_rate.num, decoder_context->format_context->streams[0]->r_frame_rate.den,
+                decoder_context->format_context->streams[video_stream_index]->avg_frame_rate.num, decoder_context->format_context->streams[0]->avg_frame_rate.den);
+        }
+
+        int enc_calc_frame_duration = 0;
         if (encoder_context->format_context->streams[0]->avg_frame_rate.num != 0 &&
             decoder_context->stream[video_stream_index]->time_base.num != 0) {
-            encoder_context->calculated_frame_duration =
+
+            encoder_context->calculated_frame_duration = enc_calc_frame_duration =
                 /* In very rare cases this might overflow, so type cast to 64bit int to avoid overflow */
```

Comment:
We're just dividing timescale by framerate right? Would be good add this in a comment because it's very difficult to read with the whole context->stream[array access] everywhere

Reply:
I changed to use `av_q_` functions but I will refactor this into a separate function in a future PR